### PR TITLE
Adds profile link and UID to Zendesk users

### LINKED ISSUE #3565: pull in link to user profile and uid into Zendesk
When a user submits a a help ticket, the link to the user's profile and the user's uid should be easily accessible on the zendesk ticket.

### PR DESCRIPTION
**Pull request by itsjoekent**
Fixes #3565

@aaronschachter

### CODE REVIEW

File: lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module [hunk 1]
```diff
@@ -351,6 +351,18 @@ function dosomething_zendesk_verify_zendesk_user($email, $client = NULL) {
...
+    $profile_fields = dosomething_zendesk_get_profile_fields($uid);
+    $id = $search->results[0]->id;
+    $updated_user = array(
+      'id' => $id,
+      'user_fields' => array(
+        'profile_url' => $profile_fields['profile_url'],
+        'profile_uid' => $profile_fields['profile_uid'],        
+      ),
+    );
+    $client->users()->update($updated_user);
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module:367]: Can we check to see if the ZD User already has the fields set, vs setting every time? These values will never change, so its unnecessary overhead to post the update every time someone submits the form.

File: lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module [hunk 2]
```diff
@@ -351,6 +351,20 @@ function dosomething_zendesk_verify_zendesk_user($email, $client = NULL) {
       // Create a zendesk user.
       return dosomething_zendesk_create_zendesk_user($email, $client);
     }
+    // Update user with profile fields if missing.
+    if (!$search->results[0]->user_fields->profile_url) {
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module]: Hmm. What does this look like if not set, does it throw an error? Or does each user return empty value here? (Do you need to check for `isset`)

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module]: Just tested it and it worked fine but it did add an error to the log 

Notice: Undefined property: stdClass::$profile_url in dosomething_zendesk_verify_zendesk_user() 

Ill change it to isset and see what happens

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_zendesk/dosomething_zendesk.module]: isset works fine pushing change now

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**